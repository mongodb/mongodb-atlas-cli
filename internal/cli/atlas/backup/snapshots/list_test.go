// Copyright 2020 MongoDB Inc
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

//go:build unit

package snapshots

import (
	"bytes"
	"testing"

	"github.com/golang/mock/gomock"
	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
	"github.com/mongodb/mongodb-atlas-cli/internal/pointer"
	"github.com/mongodb/mongodb-atlas-cli/internal/test"
	atlasv2 "go.mongodb.org/atlas-sdk/v20231115002/admin"
)

func TestList_Run(t *testing.T) {
	ctrl := gomock.NewController(t)
	mockStore := mocks.NewMockSnapshotsLister(ctrl)
	buf := new(bytes.Buffer)
	expected := &atlasv2.PaginatedCloudBackupReplicaSet{
		Results: []atlasv2.DiskBackupReplicaSet{
			{
				CloudProvider: pointer.Get("AWS"),
				Id:            pointer.Get("5f9b0b5e0b5e9d6b6e0b5e9d"),
				SnapshotType:  pointer.Get("cloud"),
			},
			*atlasv2.NewDiskBackupReplicaSet(),
		},
	}

	listOpts := &ListOpts{
		store:       mockStore,
		clusterName: "Cluster0",
		OutputOpts: cli.OutputOpts{
			Template:  listTemplate,
			OutWriter: buf,
		},
	}

	mockStore.
		EXPECT().
		Snapshots(listOpts.ProjectID, "Cluster0", listOpts.NewListOptions()).
		Return(expected, nil).
		Times(1)

	if err := listOpts.Run(); err != nil {
		t.Fatalf("Run() unexpected error: %v", err)
	}
	test.VerifyOutputTemplate(t, listTemplate, expected)
}
