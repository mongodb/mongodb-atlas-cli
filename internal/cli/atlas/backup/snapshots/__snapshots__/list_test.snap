
[TestList_Run - 1]
ID                         TYPE    STATUS   CREATED AT   EXPIRES AT
5f9b0b5e0b5e9d6b6e0b5e9d   cloud   <nil>    <nil>        <nil>
<nil>                      <nil>   <nil>    <nil>        <nil>

---
